python_modules/dagster/dagster/core/definitions/metadata/__init__.py

```diff
@@ -485,23 +485,17 @@ def __new__(cls, url: Optional[str]):
 
 @whitelist_for_serdes(storage_name="PathMetadataEntryData")
 class PathMetadataValue(  # type: ignore
-    NamedTuple(
-        "_PathMetadataValue",
-        [
-            ("path", Optional[str]),
-        ],
-    ),
-    MetadataValue,
+    NamedTuple("_PathMetadataValue", [("path", Optional[Union[str, os.PathLike]])]), MetadataValue
```

Comment:
`NamedTuple` fields shouldn't changes-- we're still going to store the `path` as a string

Reply:
Internal storage vs what it accepts, got it. I largely was just propagating typing downwards.